ecc/bls12-377/fr/pedersen/pedersen.go

```diff
@@ -273,37 +271,46 @@ func (vk *VerifyingKey) Verify(commitment curve.G1Affine, knowledgeProof curve.G
 // must be randomly generated by the verifier and sent to the prover. Otherwise,
 // it must be generated via Fiat-Shamir.
 func BatchVerifyMultiVk(vk []VerifyingKey, commitments []curve.G1Affine, pok []curve.G1Affine, combinationCoeff fr.Element) error {
-	if len(commitments) != len(vk) || len(pok) != len(vk) {
-		return errors.New("length mismatch")
+	if len(commitments) != len(vk) {
+		return errors.New("commitments length mismatch")
 	}
-	for i := range pok {
-		if !pok[i].IsInSubGroup() || !commitments[i].IsInSubGroup() {
-			return errors.New("subgroup check failed")
+	// we use folded POK if provided
```

Comment:
Documentation should mention this. Perhaps it'd be better to put `pok` at the end of the argument list and make it variadic?

Reply:
Added to the documentation.

I'd preferably avoid using variadic arguments unless necessary. This allows in the future to add options to the methods in case we need it.